### PR TITLE
spdlog: fix cross-build to Android + modernize

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

See https://cpplang.slack.com/archives/C41CWV9HA/p1639564662279500

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SSE4** [comment]
the diff is quite a big, can you highlight what's the actual fix for the Android amongst all the changes?

**Comment by SpaceIm** [comment]
https://github.com/conan-io/conan-center-index/pull/8441/commits/6c4f0970204d1a4b73334c4bee6c745f1a073812

File: recipes/spdlog/all/conanfile.py [hunk 1]
```diff
@@ -109,28 +126,27 @@ def package(self):
...
 
-    def package_id(self):
-        if self.options.header_only:
-            self.info.header_only()
-
     def package_info(self):
-        if self.options.header_only:
-            component_name = "spdlog_header_only"
-        else:
-            component_name = "libspdlog"
```
Comment (ericLemanissier) [inline recipes/spdlog/all/conanfile.py:120]: This is a breaking change. consumers could have components requiring spdlog::spdlog_header_only.

Comment (SpaceIm) [reply recipes/spdlog/all/conanfile.py:120]: Yes it's a breaking change, conditional component name is bad. This component will even disapear from this recipe in conan v2, it was a side effect of `.names` mechanism.

**Review by madebr — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — COMMENTED** [review]
LGTM

**Review by uilianries — APPROVED** [review]

**Pull request merged**